odoo/models.py

```diff
@@ -2446,6 +2450,29 @@ def _inherits_join_add(self, current_model, parent_model_name, query):
         )
         return parent_alias
 
+    def _join_calc(self, alias, fname, query):
```

Comment:
I didn't want to pollute too much of the ORM, I figured that since the goal is to only allow grouping by m2m inside read_group, I should limit the scope to that as much as possible.

With the wrapped method, it only affects `read_group` since it's the only method to use `_join_calc`, however if I were to change `_inherits_join_calc` to accept m2m fields, tons of callsites would be able to use a feature (joining by m2m field) that is really only intended for `read_group`, I don't know the implications of that.

But just let me know and I'll change it to simply extend `_inherits_join_calc`

Reply:
PS: this could be another case in which we could delegate the join implementation shenanigans to the field, but it could be much less useful here since I think only many2many fields would have a different implementation from all other fields